### PR TITLE
add s3dis automatic download

### PR DESCRIPTION
**Pull request by tchaton**

### CODE REVIEW

**Comment by tchaton** [comment]
Screenshot 2020-08-08 at 17 43 20
Screenshot 2020-08-08 at 19 32 08

File: test/test_random_cylinder.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+import torch
+from torch_geometric.data import Data
+
+ROOT = os.path.join(os.path.dirname(os.path.realpath(__file__)), "..")
+sys.path.insert(0, ROOT)
+
+from torch_points3d.core.data_transform.transforms import CylinderSampling
+
+
+class TestRandmSphere(unittest.TestCase):
```
Comment (nicolas-chaulet) [inline test/test_random_cylinder.py:14]: Please rename to RandomCylinder

File: torch_points3d/core/data_transform/transforms.py [hunk 2]
```diff
@@ -270,6 +337,89 @@ def __repr__(self):
...
+class CylinderNormalizeScale(object):
+    """ Normalize points within a cylinder
+
+    """
+    def __init__(self, normalize_z=True):
+        self._normalize_z = normalize_z
+
+    def _process(self, data):
+        data.pos -= data.pos.mean(dim=0, keepdim=True)
+        scale = (1 / data.pos[:, :-1].abs().max()) * 0.999999
```
Comment (nicolas-chaulet) [inline torch_points3d/core/data_transform/transforms.py:405]: why 0.99999?

File: torch_points3d/core/data_transform/transforms.py [hunk 3]
```diff
@@ -270,6 +337,89 @@ def __repr__(self):
...
+        data.pos[:, :-1] = data.pos[:, :-1] * scale
+        if self._normalize_z:
+            scale = (1 / data.pos[:, -1].abs().max()) * 0.999999
```
Comment (nicolas-chaulet) [inline torch_points3d/core/data_transform/transforms.py:408]: Same

File: torch_points3d/core/data_transform/transforms.py [hunk 4]
```diff
@@ -270,6 +337,89 @@ def __repr__(self):
...
+
+    """
+    def __init__(self, normalize_z=True):
+        self._normalize_z = normalize_z
+
+    def _process(self, data):
+        data.pos -= data.pos.mean(dim=0, keepdim=True)
+        scale = (1 / data.pos[:, :-1].abs().max()) * 0.999999
+        data.pos[:, :-1] = data.pos[:, :-1] * scale
+        if self._normalize_z:
```
Comment (nicolas-chaulet) [inline torch_points3d/core/data_transform/transforms.py:407]: Looks like you normalize in any case

File: torch_points3d/datasets/panoptic/s3dis.py [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+    S3DISCylinder,
+    add_weights, 
+    INV_OBJECT_LABEL
+)
+import torch_points3d.core.data_transform as cT
+from torch_points3d.metrics.panoptic_tracker import PanopticTracker
+from torch_points3d.datasets.panoptic.utils import set_extra_labels
+
+STUFF_CLASSES_INV = {
+    # 0: "ceiling",
```
Comment (nicolas-chaulet) [inline torch_points3d/datasets/panoptic/s3dis.py:17]: There is no stuff class for s3dis in PointGroup. All classes are used for panoptic segmentation

File: torch_points3d/datasets/panoptic/s3dis.py [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+
+    def _set_extra_labels(self, data):
+        return set_extra_labels(data, self.INSTANCE_CLASSES, self.NUM_MAX_OBJECTS, self._remap_labels)
+
+    def _remap_labels(self, semantic_label):
+        return semantic_label
+
+    @property
+    def stuff_classes(self):
+        return super()._remap_labels(self.INSTANCE_CLASSES)
```
Comment (nicolas-chaulet) [inline torch_points3d/datasets/panoptic/s3dis.py:60]: Empty tensor

File: torch_points3d/datasets/panoptic/utils.py [hunk 7]
```diff
@@ -0,0 +1,52 @@
...
+    if num_instances:
+        data.center_label[:num_instances, :] = torch.stack(centers)
+
+    data.vote_label = point_votes.float()
+    data.instance_labels = instance_labels
+    data.instance_mask = instance_labels != 0
+    data.num_instances = torch.tensor([num_instances])
+
+    # Remap labels
+    data.y = remap_labels(data.y)
```
Comment (nicolas-chaulet) [inline torch_points3d/datasets/panoptic/utils.py:51]: This remap_labels should be called from the dataset itself, it is too specific and having it here hides it a bit.

File: torch_points3d/datasets/panoptic/scannet.py [hunk 8]
```diff
@@ -41,54 +42,7 @@ def __getitem__(self, idx):
...
-
-        data.vote_label = point_votes.float()
-        data.instance_labels = instance_labels
-        data.instance_mask = instance_labels != 0
-        data.num_instances = torch.tensor([num_instances])
-
-        # Remap labels
-        data.y = super()._remap_labels(data.y)
-        return data
+        return set_extra_labels(data, self.NYU40ID2CLASS, self.NUM_MAX_OBJECTS, self._remap_labels)
```
Comment (nicolas-chaulet) [inline torch_points3d/datasets/panoptic/scannet.py:45]: call set_extra_labels and _remap_labels just after.

File: torch_points3d/datasets/panoptic/s3dis.py [hunk 9]
```diff
@@ -0,0 +1,150 @@
...
+        data = super().__getitem__(idx)
+
+        # Extract instance and box labels
+        self._set_extra_labels(data)
+        return data   
+
+    def _set_extra_labels(self, data):
+        return set_extra_labels(data, self.INSTANCE_CLASSES, self.NUM_MAX_OBJECTS, self._remap_labels)
+
+    def _remap_labels(self, semantic_label):
```
Comment (nicolas-chaulet) [inline torch_points3d/datasets/panoptic/s3dis.py:55]: No need for this dataset

**Comment by tchaton** [comment]
Screenshot 2020-08-12 at 19 04 09

File: torch_points3d/datasets/panoptic/s3dis.py [hunk 10]
```diff
@@ -50,14 +50,11 @@ def __getitem__(self, idx):
...
-        return set_extra_labels(data, self.INSTANCE_CLASSES, self.NUM_MAX_OBJECTS, self._remap_labels)
-
-    def _remap_labels(self, semantic_label):
-        return semantic_label
+        return set_extra_labels(data, self.INSTANCE_CLASSES, self.NUM_MAX_OBJECTS)
 
     @property
     def stuff_classes(self):
-        return super()._remap_labels(self.INSTANCE_CLASSES)
+        return torch.tensor([i for i in self.INSTANCE_CLASSES.keys()])
```
Comment (nicolas-chaulet) [inline torch_points3d/datasets/panoptic/s3dis.py:57]: This should be an empty tensor. Stuff stands for things we don't care about.
